code/hsec-tools/src/Security/Advisories/Filesystem.hs

```diff
@@ -129,10 +130,12 @@ listAdvisories root =
       then return $ pure []
       else do
         oob <-
-          liftIO (getAdvisoryGitInfo advisoryPath) <&> \case
-            Left _ -> emptyOutOfBandAttributes
+          liftIO (getAdvisoryGitInfo advisoryPath) >>= \case
+            Left gitErr -> 
+              liftIO (hPutStrLn stderr ("obtaining out of band attributes failed: \n" <> explainGitError gitErr))
```

Comment:
May we return a validation error instead of a warning on the stderr?

Reply:
chesterton's fence: why's the error ignoed in the first place, are there valid reasons to not abort here?

Reply:
I think the idea is that we don't want to fail if there's no git information as there might still be one supplied by the advisory itself. I think this is an issue with the structure of the code.

Reply:
I have decided to bubble the error up instead of using this `oobValue` with optional fields. I think it was just not very fortunately designed.

Reply:
Thanks, that looks reasonable to me. Though I worry this is a complicated API for downstream users, perhaps we could do instead:
- Make the oob attributes explicit in the advisory document, and/or
- Publish a static version of the advisories with the oob attributes pre resolved.

What do you think @frasertweedale ?

Reply:
Publishing a version with the oob pre resolved would be a good idea anyways. That would make testing without git possible and generally would allow distribution without git. I don’t think the interface for downstream users is more complicated now, tbh it was quite complicated with these fields that could independently be Nothing when in reality I don’t see a reason why they should be able to. Not attaching a reason for why the oob attributes being missing was problematic as well, imo, this caused the Git errors to never be shown.